src/Elliptic/Protocols/FirstOrderSystem.hpp

```diff
@@ -0,0 +1,207 @@
+      "other. In particular, each primal flux must have one "
+      "index more than its corresponding primal field and an upper-spatial "
+      "first index.");
+  static_assert(
+      ((tmpl::size<typename AuxiliaryFluxes::type::index_list>::value ==
+        tmpl::size<typename AuxiliaryFields::type::index_list>::value + 1) and
+       ...) and
+          (std::is_same_v<
+               tmpl::front<typename AuxiliaryFluxes::type::index_list>,
+               SpatialIndex<Dim, UpLo::Up, Frame::Inertial>> and
+           ...),
+      "Auxiliary fluxes and auxiliary fields must correspond to each "
+      "other. In particular, each auxiliary flux must have one "
+      "index more than its corresponding auxiliary field and an upper-spatial "
+      "first index.");
+};
+}  // namespace FirstOrderSystem_detail
+
+/*!
+ * \brief A system of elliptic equations in first-order "flux" formulation
+ *
+ * Classes conforming to this protocol represent a set of elliptic partial
+ * differential equations in first-order "flux" formulation:
+ *
+ * \f{equation}
+ * -\partial_i F^i_\alpha + S_\alpha = f_\alpha(x)
+ * \f}
+ *
+ * in terms of fluxes \f$F_\alpha^i\f$, sources \f$S_\alpha\f$ and fixed-sources
+ * \f$f_\alpha(x)\f$. It resembles closely formulations of hyperbolic
+ * conservation laws but allows the fluxes \f$F_\alpha^i\f$ to be higher-rank
+ * tensor fields. The fluxes and sources are functionals of the "primal" system
+ * variables \f$u_A(x)\f$ and their corresponding "auxiliary" variables
+ * \f$v_A(x)\f$. The fixed-sourced \f$f_\alpha(x)\f$ are independent of the
+ * system variables. We enumerate the variables with uppercase letters such that
+ * \f$v_A\f$ is the auxiliary variable corresponding to \f$u_A\f$. Greek letters
+ * enumerate _all_ variables. In documentation related to particular elliptic
+ * systems we generally use the canonical system-specific symbols for the fields
+ * in place of these indices. See the `Poisson::FirstOrderSystem` and the
+ * `Elasticity::FirstOrderSystem` for examples.
+ *
+ * Conforming classes must have these static member variables:
+ *
+ * - `size_t volume_dim`: The number of spatial dimensions.
+ *
+ * Conforming classes must have these type aliases:
+ *
+ * - `primal_fields`: A list of tags representing the primal fields. These are
+ *   the fields we solve for, e.g. \f$u\f$ for a Poisson equation.
+ *
+ * - `auxiliary_fields`: A list of tags representing the auxiliary fields, which
+ *   are typically gradients of the primal fields, e.g. \f$v_i = \partial_i u\f$
+ *   for a Poisson equation. These must follow the order of the `primal_fields`.
+ *   Specifically, each auxiliary field must have one rank higher than its
+ *   corresponding primal field.
+ *
+ * - `primal_fluxes`: A list of tags representing the primal fluxes
+ *   \f$F_{u_A}^i\f$. These are typically some linear combination of the
+ *   auxiliary fields with raised indices, e.g. \f$v^i = g^{ij}v_j\f$ for a
```

Comment:
could you define `g^{ij}` please? In the evolution code we are trying to use `g^{ij}` as the spatial part of the spacetime metric, not the spatial metric

Reply:
this is just a curved-space Poisson equation so there is no time. `g^{ij}` is just the metric of the Riemannian manifold that the Poisson eqn is formulated on. I added some docs

Reply:
so it's the spatial metric? In that case would it be better to use `gamma^{ij}`?